src/models/GitIndex.js

```diff
@@ -55,18 +56,24 @@ function parseBuffer (buffer) {
   let shaComputed = shasum(buffer.slice(0, -20))
   let shaClaimed = buffer.slice(-20).toString('hex')
   if (shaClaimed !== shaComputed) {
-    throw new Error(
-      `Invalid checksum in GitIndex buffer: expected ${shaClaimed} but saw ${shaComputed}`
-    )
+    throw new GitError(E.InternalFail, {
+      message: `Invalid checksum in GitIndex buffer: expected ${shaClaimed} but saw ${shaComputed}`
```

Comment:
GitError constructor as I undestand execute messages function that return translation of the message. Isn't you end up with `error.message` and `error.data.message`?

Reply:
Eh... true, but I don't expect a lot of users to ever encounter these InternalFail errors.